src/components/CustomRegisterField/CustomRegisterField.tsx

```diff
@@ -0,0 +1,71 @@
+import { type FC, type ChangeEventHandler, type ReactNode, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import type { GetRegisterFieldOption } from '@inplayer-org/inplayer.js';
+
+import Checkbox from '#components/Checkbox/Checkbox';
+import TextField from '#components/TextField/TextField';
+import Radio from '#components/Radio/Radio';
+import Dropdown from '#components/Dropdown/Dropdown';
+import DateField from '#components/DateField/DateField';
+import { ConsentFieldVariants, REGISTER_FIELD_VARIANT } from '#src/services/inplayer.account.service';
+import countriesCodes from '#static/countries-codes.json';
+import usStatesCodes from '#static/us-states-codes.json';
+
+type Props = {
+  type: ConsentFieldVariants;
+  name: string;
+  value: string | boolean;
+  onChange: ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
+} & Partial<{
+  label: ReactNode;
+  placeholder: string;
+  error: boolean;
+  helperText: string;
+  disabled: boolean;
+  required: boolean;
+  options: GetRegisterFieldOption;
+}>;
+
+export type CustomRegisterFieldCommonProps = Props;
+
+export const CustomRegisterField: FC<Props> = ({ type, value = '', ...props }) => {
+  const { t } = useTranslation(type);
+
+  const optionsList = useMemo(() => {
+    switch (type) {
+      case REGISTER_FIELD_VARIANT.COUNTRY_SELECT:
+      case REGISTER_FIELD_VARIANT.US_STATE_SELECT: {
+        return (type === REGISTER_FIELD_VARIANT.COUNTRY_SELECT ? countriesCodes : usStatesCodes).map((code) => ({
```

Comment:
This is kind of a predicament. As much as possible we want to define the country and state codes in 1 place and have the translations automatically pull from that place. Can you update the i18n scripts to read the json files and make sure the expected tags are automatically added to all languages?

Reply:
This is a specific case in which I have to iterate through a known list of tags before the actual "translation" occurs in runtime. There are two things I can do that can think of right now:

1. Statically import `country.json` and `us_state.json` from `#src/../public/locales/en`. Which immediately makes those files part of the bundle, even though `react-i18next` would load practically the same data dynamically. I don't think it's something either of us would like.
2. Or I can get the tag list in runtime from `const { i18n } = useTranslation()` and then `Object.keys(i18n.getResourceBundle('en', 'country'))`, but that way I stumble to another problem - the `i18n` instance is NOT initialised in test environments, so there's no `.getResourceBundle()` function available, so unit tests for the `CustomRegisterField` fail.